AK/Utf8View.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <AK/StringView.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+class Utf8View;
+
+class Utf8CodepointIterator {
+    friend class Utf8View;
+
+public:
+    ~Utf8CodepointIterator() {}
+
+    bool operator==(const Utf8CodepointIterator&) const;
+    bool operator!=(const Utf8CodepointIterator&) const;
+    Utf8CodepointIterator& operator++();
+    u32 operator*() const;
+
+private:
+    Utf8CodepointIterator(const char*, int);
+    const unsigned char* m_ptr;
+    int m_length;
+};
+
+class Utf8View {
+public:
+    explicit Utf8View(StringView);
```

Comment:
Since `StringView` is quite large (12 bytes), let's pass it by const reference.

Reply:
Yeah, I've seen that it's passed by const reference in other places, but assumed it was small enough that it doesn't matter. But if you say so, I'll change it 😀

Reply:
It used to be 8 bytes which felt small enough since it only took up 2 registers, but then we added the `StringImpl*` to it for efficient `String(StringView)` construction and now it takes up 3 registers... which starts to feel kinda chunky :)